src/sympc/tensor/mpc_tensor.py

```diff
@@ -70,16 +79,24 @@ def __init__(
                 for i, share in enumerate(self.share_ptrs):
                     if share.client == secret.client:  # type: ignore
                         self.share_ptrs[i] = self.share_ptrs[i] + secret_share
-                        break
+                        return
             else:
-                self.share_ptrs = []
+                nr_parties = len(self.session.parties)
+                tensor_type = self.session.tensor_type
+                shares = MPCTensor.generate_shares(secret, nr_parties, tensor_type)
+
+        if shares is not None and not ispointer(shares[0]):
+            shares = MPCTensor.distribute_shares(shares, self.session.parties)
```

Comment:
can you factorize this clause with L66 `if secret is not None:` ?

Reply:
Hmm...looked again into this, this is for the case we don't get a ```secret``` and get a list of shares (that are local) and we need to distribute them.